src/optimizer/query_rewrite.c

```diff
@@ -4667,7 +4672,38 @@ qo_convert_to_range (PARSER_CONTEXT * parser, PT_NODE ** wherep)
 	    }
 
 	  arg1_prior = pt_get_first_arg_ignore_prior (dnf_node);
-	  if (!pt_is_attr (arg1_prior) && !pt_is_instnum (arg1_prior))
+
+	  is_attr = true;
+	  is_all_constant = true;
+	  if (PT_IS_FUNCTION (arg1_prior) && PT_IS_SET_TYPE (arg1_prior)
+	      && arg1_prior->info.function.function_type == F_SEQUENCE)
+	    {
+	      /* multi_col_term can convert to range if arg1 is (attr,func_idx_expr,constant) */
+	      func_arg = arg1_prior->info.function.arg_list;
+	      for ( /* none */ ; func_arg; func_arg = func_arg->next)
+		{
+		  if (!pt_is_attr (func_arg) && !pt_is_function_index_expression (func_arg) && !pt_is_const (func_arg))
+		    {
+		      is_attr = false;
+		      break;
+		    }
+		  else if (!pt_is_const (func_arg))
+		    {
+		      is_all_constant = false;
```

Comment:
why is not "break" here ?

Reply:
this is check if func_arg is all constant value. so break is not needed